### PR TITLE
do not offer to change display name or password, if not possible.

### LINKED ISSUE #12319: User Management: Make "Display Name" for LDAP-Users non-editable
### Steps to reproduce
1. Login to Nextcloud as Administrator, go to Users
2. Try to change name of LDAP Users

### Expected behaviour
Field is read-only or not readable at all (just a string).
This may apply to other forms of external authentication also (e.g. OAuth, not tested).

### Actual behaviour
You can type in a new display name. 
Information seems to be not saved, on next page reload, old name apperars.

### Server configuration

**Nextcloud version:** 14.0.3

### PR DESCRIPTION
**Pull request by blizzz**
Fixes #12319

That's more a quick prototype, tests might fail, did not look for them.

Question: can we just extend the OCS endpoint with those additional data? In fact, it it could be more compact with the backend data as different result sets and users just pointing to the related backend feature set … 

Also, it does not look very consistent. There should be an indicator whether a display name is editable or not, but that should not vary too much. That's something for CSS magicians howerver :)

usermgmt

### CODE REVIEW

File: settings/src/components/userList/userRow.vue [hunk 1]
```diff
@@ -42,13 +42,15 @@
...
 		<form class="displayName" :class="{'icon-loading-small': loading.displayName}" v-on:submit.prevent="updateDisplayName">
-			<input :id="'displayName'+user.id+rand" type="text"
+			<input v-if="user.backendCapabilities.setDisplayName"
+				    :id="'displayName'+user.id+rand" type="text"
 					:disabled="loading.displayName||loading.all"
 					:value="user.displayname" ref="displayName"
 					autocomplete="new-password" autocorrect="off" autocapitalize="off" spellcheck="false" />
-			<input type="submit" class="icon-confirm" value="" />
+			<input v-if="user.backendCapabilities.setDisplayName" type="submit" class="icon-confirm" value="" />
+			<div v-if="!user.backendCapabilities.setDisplayName" class="name">{{user.displayname}}</div>
```
Comment (skjnldsv) [inline settings/src/components/userList/userRow.vue]: could use `v-else` :wink:

Comment (blizzz) [reply settings/src/components/userList/userRow.vue]: can I combine previous same v-ifs?

**Comment by johkoenig** [comment]
I really appreciate how fast you put my feature request into real code. Thumbs up for that.

> Also, it does not look very consistent. There should be an indicator whether a display name is editable or not, but that should not vary too much. That's something for CSS magicians howerver :)

What about adding a little (i) behind the display name which, when hovering over it, says "Username generated from $Backend information"?

**Comment by blizzz** [comment]
> I really appreciate how fast you put my feature request into real code. Thumbs up for that.

tbh, I think it is a long standing issue, just could not find the original bug report. Might be well from before-fork-times :D I was curious to see whether I could bake this in into the new frontend code for users management :)

> What about adding a little (i) behind the display name which, when hovering over it, says "Username generated from $Backend information"?

A valid for statement LDAP, hypothetically not for other possible backends. But perhaps a tooltip that states that the values cannot be changed against the backend would suffice I guess.

File: settings/src/components/userList/userRow.vue (hunk 1 above)
Comment (blizzz) [reply settings/src/components/userList/userRow.vue]: both aspects done

**Comment by blizzz** [comment]
Added the tooltip, updated the screenshot in the description

**Comment by johkoenig** [comment]
> tbh, I think it is a long standing issue, just could not find the original bug report. Might be well from before-fork-times :D I was curious to see whether I could bake this in into the new frontend code for users management :)
Not a problem. For me, the result's the same.

By the way, your current wording "The backend does not support changing the display name" is a little bit misleading. Despite what you say, the display name can be changed. The point is, it must be done in the backend. :-)
So, I propose something like "Display name is generated from backend data"

**Comment by blizzz** [comment]
> By the way, your current wording "The backend does not support changing the display name" is a little bit misleading. Despite what you say, the display name can be changed. The point is, it must be done in the backend. :-)

My intention is to refer to the backend in Nextcloud context, but you can argue it's a connector to the backend service. I had the same thoughts, but did not see it as much of a problem.

**Review by rullzer — APPROVED** [review]
Code makes sense

**Comment by blizzz** [comment]
need to touch users acceptance tests yet

File: settings/src/components/userList/userRow.vue [hunk 2]
```diff
@@ -42,13 +42,17 @@
...
+						:id="'displayName'+user.id+rand" type="text"
+						:disabled="loading.displayName||loading.all"
+						:value="user.displayname" ref="displayName"
+						autocomplete="new-password" autocorrect="off" autocapitalize="off" spellcheck="false" />
+				<input v-if="user.backendCapabilities.setDisplayName" type="submit" class="icon-confirm" value="" />
+			</template>
+			<div v-else class="name" v-tooltip.auto="t('settings', 'The backend does not support changing the display name')">{{user.displayname}}</div>
 		</form>
-		<form class="password" v-if="settings.canChangePassword" :class="{'icon-loading-small': loading.password}"
+		<form class="password" v-if="settings.canChangePassword && user.backendCapabilities.setPassword" :class="{'icon-loading-small': loading.password}"
```
Comment (skjnldsv) [inline settings/src/components/userList/userRow.vue]: You also need to hide the header, if you hide the full column

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: settings/src/components/userList/userRow.vue (hunk 2 above)
Comment (blizzz) [reply settings/src/components/userList/userRow.vue]: Negative. This is backend-specific and can look very different for other users (see screenshot). The else part adds there an empty div.

Comment (skjnldsv) [reply settings/src/components/userList/userRow.vue]: My bad, it was already there, just hidden a few line under on the diff :)
Thanks!

**Review by skjnldsv — APPROVED** [review]

File: settings/src/components/userList/userRow.vue (hunk 2 above)
Comment (blizzz) [reply settings/src/components/userList/userRow.vue]: no prob, thank you :)

**Comment by blizzz** [comment]
Okay, after wondering why it failed… it just passes locally. 

acceptance

→ merging

I hate those CI hiccups :(

**Pull request merged**